plugins/module_utils/fusion.py

```diff
@@ -100,39 +101,42 @@ def get_fusion(module):
 
     issuer_id = module.params[PARAM_ISSUER_ID]
     key_file = module.params[PARAM_PRIVATE_KEY_FILE]
+    access_token = module.params[PARAM_ACCESS_TOKEN]
 
     config = fusion.Configuration()
     config.host = environ.get(ENV_API_HOST, environ.get(ENV_HOST, config.host))
     config.token_endpoint = environ.get(ENV_TOKEN_ENDPOINT, config.token_endpoint)
-
-    if issuer_id is not None and key_file is not None:
-        try:
-            config.issuer_id = issuer_id
-            config.private_key_file = key_file
-            client = fusion.ApiClient(config)
-            client.set_default_header("User-Agent", user_agent)
-        except Exception:
-            module.fail_json(msg="Unknown failure. Please contact Pure Support")
+    
+    if access_token is not None:
+        config.access_token = access_token
+    elif issuer_id is not None and key_file is not None:
+        config.issuer_id = issuer_id
+        config.private_key_file = key_file
+    elif ENV_ACCESS_TOKEN in environ:
+        config.access_token = environ.get(ENV_ACCESS_TOKEN)
     elif (
         ENV_ISSUER_ID in environ or ENV_APP_ID in environ
     ) and ENV_PRIVATE_KEY_FILE in environ:
-        try:
-            config.issuer_id = environ.get(ENV_ISSUER_ID, environ.get(ENV_APP_ID))
-            config.private_key_file = environ.get(ENV_PRIVATE_KEY_FILE)
-            client = fusion.ApiClient(config)
-            client.set_default_header("User-Agent", user_agent)
-        except Exception:
-            module.fail_json(msg="Unknown failure. Please contact Pure Support")
+        config.issuer_id = environ.get(ENV_ISSUER_ID, environ.get(ENV_APP_ID))
+        config.private_key_file = environ.get(ENV_PRIVATE_KEY_FILE)
     else:
         module.fail_json(
-            msg=f"You must set {ENV_ISSUER_ID} and f{ENV_PRIVATE_KEY_FILE} environment variables "
-            f"or the {PARAM_ISSUER_ID} and f{PARAM_PRIVATE_KEY_FILE} module arguments"
+            msg=f"You must set either {ENV_ISSUER_ID} and f{ENV_PRIVATE_KEY_FILE} or {ENV_ACCESS_TOKEN} environment variables. "
+            f"Or module arguments either {PARAM_ISSUER_ID} and f{PARAM_PRIVATE_KEY_FILE} or {ENV_ACCESS_TOKEN}"
         )
+        
+    try:
+        client = fusion.ApiClient(config)
+        client.set_default_header("User-Agent", user_agent)
+    except Exception:
+            module.fail_json(msg="Unknown failure. Please contact Pure Support")
+        
```

Comment:
Can we merge these two try...except blocks? It would be a good opportunity to add exception details to the return message, possibly as an extra field so the user gets some details about what happened and can unblock himself

Reply:
e.g. module.fail_json(msg="...", details=err)

Reply:
Agree with @jankodera-ps

Reply:
what the second except block?

Reply:
you are talking about this one?
```
try:
        api_instance = fusion.DefaultApi(client)
        api_instance.get_version()
    except Exception as err:
        module.fail_json(msg="Fusion authentication failed: {0}".format(err))
```

Reply:
yes

Reply:
@andreepyro  But we should have different message here. When we fail to create client by config, it is unknown problem. But if it was created successfully, but we cannot make API call in `Fusion authentication error`

Reply:
You can do something like this:
```
try:
   ...
except purefusion.rest.ApiException as err:
   ... # handle api error
except HTTPError as err:
   ... # handle http error
except Exception as err:
   ...  # handle other errors
```
As far as I know all API calls can produce only `purefusion.rest.ApiException` or 'HTTPError'.

Not sure whether we need to strictly differentiate between the authentication phase and instance creation phase @jankodera-ps ?

Reply:
Now when I think bout it `purefusion.rest.ApiException` should be already handled by our hooks. So checking only for general `Exception` might be enough.

Reply:
My main peeve was that previously there were no details in the error if APIClient creation failed, just something in line of `Please contant Pure Support.`, which is as hostile for the customer as it gets. Merging it with an error below was a way to introduce more sensible error so the user can help himself if possible. I've looked into the Python SDK and there do not seem to be any hot path which could raise an exception (apart from coding bugs), so this error is okay.